hw/ip/edn/doc/_index.md

```diff
@@ -0,0 +1,171 @@
+- A set of registers is provided for firmware to manage a CSRNG application port.
+- Hardware interfaces support four request/acknowledge interfaces, or can be configured to act as a broadcast type of interface (only 1 of 4 interfaces).
+- Hardware interface supports a parameterized data delivery width, for powers of 2 from 1 to 128-bit widths.
+- Auto request mode where generate and reseed commands can be programmed to be done continuously in hardware.
+- Boot mode where a basic instantiate and generate command are done with one TLUL configuration write.
+- Interrupts that are supported:
+  - CSRNG application command has completed.
+  - An internal FIFO error has occurred.
+- No alerts are supported by this revision.
+
+
+## Description
+
+This IP block acts as a gasket between requesting peripheral blocks and the CSRNG block.
+One function this IP block performs is to translate data transfer size.
+For example, CSRNG will return 128 bits on the `genbits` bus.
+A peripheral block may connect with a 32-bit data bus.
+The EDN block will move the first 32 bits from the returned `genbits` bus, and hold on to the remaining data until another request ask for more data. 
+Further, if data is not consumed immediately, the interface to the CSRNG will indicate back pressure to the CSRNG block.
+Each of the four interfaces can request data such that any `genbits` bus return can supply any requesting peripheral block.
+
+A configurable feature for peripheral port 0 is that it can be put into broadcast mode.
+This means that the EDN block will request for that port and is paced by a configurable timer.
+Data will be delivered continuously on the bus for any snooping participant to consume.
+
+Commands to the CSRNG block can be generated by either firmware or hardware.
+For firmware generated commands, a TLUL set of registers will allow any CSRNG application interface command to be sent to CSRNG.
+For hardware generated commands, there are two modes that allow this.
+One is the auto request mode, where two FIFOs are provided to send commands and additional data. 
+The general operation of this mode is that the CSRNG instance is set up by firmware, then the FIFOS are preloaded with commands.
+One FIFO can be programmed to send `generate` commands.
+The other FIFO can be programmed to send `reseed` commands.
+There is a configurable register that sets the number of requests between reseeds.
+This register determines how many `generate` commands are sent before a `reseed` command is sent.
+The process of sending these two commands will repeat forever until the module is reset or the enable bit is cleared.
+
+## Example Topology
+The diagram below shows an example topology where two EDN modules are used to distribute CSRNG `genbits` to peripheral modules.
+
+![EDN Example Topology Diagram](edn_top_diag.svg)
+
+
+## Compatibility
+Commands generated toward the CSRNG application interface must conform to the CSRNG application interface specification.
```

Comment:
I'm unconvinced that this text is needed! Can't this document just say that writes to the relevant registers are forwarded to CSRNG. What happens after that is up to the CSRNG spec, and the EDN doesn't need to care about it.

Reply:
Right.  Well it's not quite as simple as "forwarding registers".  There is a pretty non-trivial command structure to CSRNG.   But I'm happy to pull this, because it's not like this some Industry standard protocol or something.  (Except insofar as CSRNG's interface "envelope" is constrained by SP 800-90A.)

Reply:
Removing this sentence.